components/infobox/wikis/starcraft2/infobox_person_player.lua

```diff
@@ -0,0 +1,541 @@
+	local name = args.id or self.pagename
+
+	return raceIcon .. '&nbsp;' .. name
+end
+
+function StarCraft2Player._getRaceData(race)
+	race = string.lower(race)
+	race = CleanRace[race] or race
+	local display = _RACE_CATEGORY[race]
+	if not display and race ~= 'unknown' then
+		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
+			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
+	end
+
+	raceData = {
+		race = race,
+		faction = _FACTION1[race] or '',
+		faction2 = _FACTION2[race] or '',
+		display = display
+	}
+end
+
+function StarCraft2Player:createBottomContent(infobox)
+	if shouldStoreData then
+		return tostring(Matches._get_ongoing({})) ..
+			tostring(Matches._get_upcoming({})) ..
+			tostring(Matches._get_recent({}))
+	end
+end
+
+function StarCraft2Player:shouldStoreData(args)
+	if
+		args.disable_smw == 'true' or args.disable_lpdb == 'true' or args.disable_storage == 'true'
+		or Variables.varDefault('disable_SMW_storage', 'false') == 'true'
+		or mw.title.getCurrentTitle().nsText ~= ''
+	then
+		Variables.varDefine('disable_SMW_storage', 'true')
+		return false
+	end
+	return true
+end
+
+function StarCraft2Player:getAchievements(infobox, args)
+	local player = args.id or self.pagename
+	local achievements = Achievements({}, player)
+	if achievements == '' then
+		achievements = nil
+	else
+		hasAchievements = true
+	end
+	return achievements
+end
+
+--kick the default history display
+--because we will add it a bit lower again
+function StarCraft2Player:getHistory()
+	return nil
+end
+
+function StarCraft2Player:addCustomContent(infobox, args)
+	local retired
+	--only display retired if it contains a year
+	if string.match(args.retired or '', '%d%d%d%d%') then
+		retired = args.retired
+	end
+	local allkills
+	if shouldStoreData then
+		allkills = StarCraft2Player._getAllkills()
+		if allkills then
+			allkills = _ALLKILLICON .. allkills
+		end
+	end
+	infobox:header('Achievements', (not hasAchievements) and allkills or nil)
+	infobox:cell('All-kills', allkills)
+	infobox:header('History', args.history)
+	infobox:centeredCell(args.history)
+	infobox:cell('Retired', retired)
+
+	return infobox
+end
+
+function StarCraft2Player._getAllkills()
+	if shouldStoreData then
+		local allkillsData = mw.ext.LiquipediaDB.lpdb('datapoint', {
+			conditions = '[[pagename::' .. pagename .. ']] AND [[type::allkills]]',
+			query = 'information',
+			limit = 1
+		})
+		if type(allkillsData[1]) == 'table' then
+			return allkillsData[1].information
+		end
+	end
+	return nil
+end
+
+function StarCraft2Player:getRole(args)
+	local role = (args.role or args.occupation or 'player'):lower()
+	local category = _ROLES[role]
+	local store = category or _CLEAN_OTHER_ROLES[role] or 'Player'
+
+	return { title = 'Race', display = raceData.display, store = store, category = category or 'Player'}
+end
+
+function StarCraft2Player:calculateEarnings(args)
+	shouldStoreData = Player.shouldStoreData(args)
```

Comment:
1) Class variable `shouldStoreData` should be set in the constructor not in a function not related to storing data.
2) Should be `self:shouldStoreData` not `Player.`